src/main/java/com/amazon/dlic/auth/http/jwt/HTTPExtensionJwtAuthenticator.java

```diff
@@ -0,0 +1,282 @@
+        this.signingKey = signingKey;
+        this.encryptionKey = encryptionKey;
+        this.BWCMode = BWCMode;
+        init();
+    }
+
+    private void init() {
+        JwtParser _jwtParser = null;
+
+        try {
+            if(signingKey == null || signingKey.length() == 0) {
+                log.error("signingKey must not be null or empty. JWT authentication will not work");
+            } else {
+
+                signingKey = signingKey.replace("-----BEGIN PUBLIC KEY-----\n", "");
+                signingKey = signingKey.replace("-----END PUBLIC KEY-----", "");
+
+                byte[] decoded = Decoders.BASE64.decode(signingKey);
+                Key key = null;
+
+                try {
+                    key = getPublicKey(decoded, "RSA");
+                } catch (Exception e) {
+                    log.debug("No public RSA key, try other algos ({})", e.toString());
+                }
+
+                try {
+                    key = getPublicKey(decoded, "EC");
+                } catch (Exception e) {
+                    log.debug("No public ECDSA key, try other algos ({})", e.toString());
+                }
+
+                if(key != null) {
+                    _jwtParser = Jwts.parser().setSigningKey(key);
+                } else {
+                    _jwtParser = Jwts.parser().setSigningKey(decoded);
+                }
+
+            }
+        } catch (Throwable e) {
+            log.error("Error creating JWT authenticator. JWT authentication will not work", e);
+            throw new RuntimeException(e);
+        }
+
+        subjectKey = "sub";
+        //TODO: requireIssuer should be the cluster name. There is one authentication backend for all extensions, so there will be multiple audiences.
+//        requireAudience = "true";
+//        requireIssuer = "true";
+//
+//        _jwtParser.requireAudience(requireAudience);
+//        _jwtParser.requireIssuer(requireIssuer);
+
+        jwtParser = _jwtParser;
+    }
+
+    @Override
+    @SuppressWarnings("removal")
+    public AuthCredentials extractCredentials(RestRequest request, ThreadContext context) throws OpenSearchSecurityException {
+        final SecurityManager sm = System.getSecurityManager();
+
+        if (sm != null) {
+            sm.checkPermission(new SpecialPermission());
+        }
+
+        AuthCredentials creds = AccessController.doPrivileged(new PrivilegedAction<AuthCredentials>() {
+            @Override
+            public AuthCredentials run() {
+                return extractCredentials0(request);
+            }
+        });
+
+        return creds;
+    }
+
+    private AuthCredentials extractCredentials0(final RestRequest request) {
+        if (jwtParser == null) {
+            log.error("Missing Signing Key. JWT authentication will not work");
+            return null;
+        }
+
+        String jwtToken = request.header(HttpHeaders.AUTHORIZATION);
```

Comment:
Does this mean that this takes anything passed in the `Authorization` header as long as it doesn't contain the word `Basic` anywhere? In the next block if it contains the word `bearer` it will strip that out and trim the rest?